src/main/java/org/dbos/apiary/postgres/PostgresContext.java

```diff
@@ -66,19 +66,24 @@ public FunctionOutput apiaryCallFunction(String name, Object... inputs) throws E
         if (functionType.equals(ApiaryConfig.postgres) || functionType.equals(ApiaryConfig.stateless)) {
             return f.apiaryRunFunction(this, inputs);
         } else {
+            ApiarySecondaryConnection c = workerContext.getSecondaryConnection(functionType);
+            long newID = ((this.functionID + calledFunctionID.incrementAndGet()) << 4);
+            Map<String, List<String>> writtenKeys = new HashMap<>();
             try {
-                ApiarySecondaryConnection c = workerContext.getSecondaryConnection(functionType);
-                long newID = ((this.functionID + calledFunctionID.incrementAndGet()) << 4);
-                FunctionOutput fo = c.callFunction(name, workerContext, txc, service, execID, newID, inputs);
+                FunctionOutput fo = c.callFunction(name, writtenKeys, workerContext, txc, service, execID, newID, inputs);
                 secondaryWrittenKeys.putIfAbsent(functionType, new HashMap<>());
-                for (String table: fo.getWrittenKeys().keySet()) {
+                for (String table: writtenKeys.keySet()) {
```

Comment:
I think this change means that we don't need to track `writtenKeys` in FunctionOuput anymore. Clean up the interface in FunctionOutput?